docs/docs/creating-a-generic-plugin.md

```diff
@@ -0,0 +1,58 @@
+---
+title: Creating a Generic plugin
+---
+
+This section is aimed at explaining the structure of a Gatsby plugin and the files that are needed to create it.
+
+As seen in the [what is a plugin doc](/docs/what-is-a-plugin/),a plugin is a piece of software that acts as an add-on and gives your site additional functionality.
+
+They contain a file, usually in the project root, called `package.json` - this file holds various metadata relevant to the project. This file is used to give information to npm that allows it to identify the project as well as handle the project's dependencies.
+
+The process of initializing a `package.json` in a project is done by running :
+
+```shell
+  npm init
+```
+
+After running the above command a series of options are listed in the CLI interface and those you select are stored in your `package.json` which contain some [files Gatsby looks for in a Plugin](/docs/what-you-dont-need-plugins-for/)
```

Comment:
```suggestion
After running the above command a series of options are listed in the command line interface (CLI). Those you select are stored in your `package.json` which contains some of the [files Gatsby looks for in a Plugin](/docs/what-you-dont-need-plugins-for/)
```

Reply:
Is this the right link? I suspect you're trying to link to the section @gillkyle recently published?

Reply:
I fixed that. I was trying to link to /docs/docs/files-gatsby-looks-for-in-a-plugin

Reply:
Have you pushed those changes? I saw the request for review and reviewed what was in the latest commit.

Reply:
Just pushed them